compiler/x/codegen/SIMDTreeEvaluator.cpp

```diff
@@ -75,11 +75,22 @@ TR::Register* OMR::X86::TreeEvaluator::SIMDloadEvaluator(TR::Node* node, TR::Cod
    tempMR = ConvertToPatchableMemoryReference(tempMR, node, cg);
    TR::Register* resultReg = cg->allocateRegister(TR_VRF);
 
-   TR::InstOpCode::Mnemonic opCode = TR::InstOpCode::bad;
+   TR::InstOpCode::Mnemonic opCode = TR::InstOpCode::MOVDQURegMem;
+   OMR::X86::Encoding encoding = Legacy;
+
    switch (node->getSize())
       {
       case 16:
-         opCode = TR::InstOpCode::MOVDQURegMem;
+         if (cg->comp()->target().cpu.supportsAVX())
+            encoding = VEX_L128;
+         break;
+      case 32:
+         TR_ASSERT_FATAL(cg->comp()->target().cpu.supportsAVX(), "256-bit vload requires AVX");
+         encoding = VEX_L256;
+         break;
+      case 64:
+          TR_ASSERT_FATAL(cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX512F), "512-bit vload requires AVX-512");
```

Comment:
Good assertions :+1:

Could you please also change `TR_ASSERT(false)` in the default case below to make it `TR_ASSERT_FATAL` like you did in `SIMDstoreEvaluator`?